Cursory upgrade of Django to v5.  Required django-u2f to be removed, …

anadon opened the pull request:
…which will likely cause all sorts of failures.

anadon committed:
Cursory upgrade of Django to v5.  Required django-u2f to be removed, which will likely cause all sorts of failures.

anadon committed:
Ripped out U2F in code.

anadon committed:
Django 5 update issue identified.

anadon committed:
Django 5 update issue identified and further fixes.

anadon committed:
Django 5 update issue identified and further fixes.

anadon committed:
Django 5 update issue identified and further fixes.

anadon committed:
Introduced a syntax corruption from one of my tics.

p0lar-bear commented:
CI passed because the job runs `pip install -r requirements.txt` and not `uv sync`.

 The job should be updated to install uv and then run the sync.

Also we should be running `./manage.py test registration` prior to commits.

anadon committed:
Rebuilt migrations since some used pytz which was removed.

anadon committed:
Found other places using timezone.

anadon committed:
Found other places using timezone.

anadon committed:
Found other places using timezone.

anadon committed:
Found more timezone issues.

anadon committed:
Reducing misc errors.

anadon committed:
Enabling missing function in Postgres, making root user in Postgres, adding branch coverage in coveralls metrics.

anadon committed:
Enabling missing function in Postgres, making root user in Postgres, adding branch coverage in coveralls metrics.

anadon committed:
Enabling missing function in Postgres, making root user in Postgres, adding branch coverage in coveralls metrics.

anadon committed:
Removing plugins not directly used; maybe that will help with these modules where errors become manifest?

p0lar-bear reviewed (commented):
@anadon I recall in furthemore/APIS#339 that [we were warned about migrations not being optional](https://github.com/furthemore/APIS/pull/339#discussion_r1898264344) and I see a bunch of migrations being deleted here. I've also noticed that there are a lot of additions to the initial - were the migrations squashed into the initial in this case?

anadon commented:
@p0lar-bear The existing migrations were mainly useful for keeping an existing DB running.  While this would be useful for backporting changes, those migrations also used timezone formats which were removed in Django 5.  I could either have hand-edited all instances of the incorrect fields, but I chose the time-saving route and regenerated a new one.

anadon committed:
Thanks to Syfaro, better understood the issue.

anadon committed:
Tweaking the db mitrations for Postgres Trigrams.

LiruMouse commented on fm_eventmanager/urls.py:
Why not just use `path` here? I don't see the point of these being regular expressions?

p0lar-bear replied:
Probably holding over from the original version to be safe.

I changed these all as suggested to use `path()` and it's working, so I'll push that.

p0lar-bear committed:
#24: Change use of re_path in root urls.py to path

p0lar-bear committed:
#24: Update README local dev instructions

p0lar-bear committed:
#24: Updating django GitHub workflow to use uv, fixing some oddities in tests.

p0lar-bear committed:
#24: Attempt to fix broken dependencies in workflow.

anadon committed:
Fixed registration.tests.test_webhooks.TestSquareRefundWebhooks.test_square_webhook_invalid_signature by adding an alternative codepath to accomplish the same function.

p0lar-bear committed:
#24: Fix typo in test_model.py

p0lar-bear committed:
#24: Revert changes to example envrc secrets file.

p0lar-bear committed:
#24: Fixing some broken tests.

Admin environment banner expects HttpRequest to have a user defined, and admin
HTML output changed the content element from a div to main.

TestTempToken now extends AdminTestCase, and I added the creation of an admin
user to AdminTestCase's setup.

p0lar-bear committed:
#24: Attempt to fix coverage step in GitHub workflow.

p0lar-bear commented:
CI is passing once again. There's just local tests breaking when running against a SQLite database. Kinda wanna take a closer look at the migrations and see about putting them back and making them work with Django 5.

p0lar-bear committed:
Revert "Rebuilt migrations since some used pytz which was removed."

This reverts commit 53c84be602cb39d33a5322c27292a17153c7c1da.

p0lar-bear commented:
> @p0lar-bear The existing migrations were mainly useful for keeping an existing DB running. While this would be useful for backporting changes, those migrations also used timezone formats which were removed in Django 5. I could either have hand-edited all instances of the incorrect fields, but I chose the time-saving route and regenerated a new one.

@anadon Was this the only factor in deciding to squash the migrations? Because there were only two files using `pytz` and each called `localize` only once to put a default time value into a variable.

In my working tree I added the `tzdata` package to the project and made changes resulting in this patch:

```
diff --git a/registration/migrations/0004_auto_20160729_0021.py b/registration/migrations/0004_auto_20160729_0021.py
index 930c28f..f729f12 100644
--- a/registration/migrations/0004_auto_20160729_0021.py
+++ b/registration/migrations/0004_auto_20160729_0021.py
@@ -1,18 +1,18 @@
 # -*- coding: utf-8 -*-
 # Generated by Django 1.9.8 on 2016-07-29 00:21
+# Modified for Django 5.x on 2026-01-11
 
 
 from datetime import datetime
 
-import pytz
+from zoneinfo import ZoneInfo
 from django.db import migrations, models
 
-PT = pytz.timezone("US/Pacific")
 
 
 class Migration(migrations.Migration):
 
-    DEFAULT_ORDER_MODIFIED_DATE = PT.localize(datetime.now())
+    DEFAULT_ORDER_MODIFIED_DATE = datetime.now(ZoneInfo("US/Pacific"))
 
     dependencies = [
         ("registration", "0003_auto_20160727_0114"),
diff --git a/registration/migrations/0034_auto_20161227_1230.py b/registration/migrations/0034_auto_20161227_1230.py
index da42fbb..a6dedf2 100644
--- a/registration/migrations/0034_auto_20161227_1230.py
+++ b/registration/migrations/0034_auto_20161227_1230.py
@@ -1,18 +1,17 @@
 # -*- coding: utf-8 -*-
 # Generated by Django 1.9.8 on 2016-12-27 17:30
+# Modified for Django 5.x on 2026-01-11
 
 
 import datetime
 
-import pytz
+from zoneinfo import ZoneInfo
 from django.db import migrations, models
 
-PT = pytz.timezone("US/Pacific")
-
 
 class Migration(migrations.Migration):
 
-    DEFAULT_DATETIME = PT.localize(datetime.datetime.now())
+    DEFAULT_DATETIME = datetime.datetime.now(ZoneInfo("US/Pacific"))
 
     dependencies = [
         ("registration", "0033_discount_used"),
```

Also on a sidenote, `influxdb` depends on `pytz` so despite removing it from the project's dependencies, it persists.

anadon commented:
> > @p0lar-bear The existing migrations were mainly useful for keeping an existing DB running. While this would be useful for backporting changes, those migrations also used timezone formats which were removed in Django 5. I could either have hand-edited all instances of the incorrect fields, but I chose the time-saving route and regenerated a new one.
> 
> @anadon Was this the only factor in deciding to squash the migrations? Because there were only two files using `pytz` and each called `localize` only once to put a default time value into a variable.
> 
> In my working tree I added the `tzdata` package to the project and made changes resulting in this patch:
> 
> ```
> diff --git a/registration/migrations/0004_auto_20160729_0021.py b/registration/migrations/0004_auto_20160729_0021.py
> index 930c28f..f729f12 100644
> --- a/registration/migrations/0004_auto_20160729_0021.py
> +++ b/registration/migrations/0004_auto_20160729_0021.py
> @@ -1,18 +1,18 @@
>  # -*- coding: utf-8 -*-
>  # Generated by Django 1.9.8 on 2016-07-29 00:21
> +# Modified for Django 5.x on 2026-01-11
>  
>  
>  from datetime import datetime
>  
> -import pytz
> +from zoneinfo import ZoneInfo
>  from django.db import migrations, models
>  
> -PT = pytz.timezone("US/Pacific")
>  
>  
>  class Migration(migrations.Migration):
>  
> -    DEFAULT_ORDER_MODIFIED_DATE = PT.localize(datetime.now())
> +    DEFAULT_ORDER_MODIFIED_DATE = datetime.now(ZoneInfo("US/Pacific"))
>  
>      dependencies = [
>          ("registration", "0003_auto_20160727_0114"),
> diff --git a/registration/migrations/0034_auto_20161227_1230.py b/registration/migrations/0034_auto_20161227_1230.py
> index da42fbb..a6dedf2 100644
> --- a/registration/migrations/0034_auto_20161227_1230.py
> +++ b/registration/migrations/0034_auto_20161227_1230.py
> @@ -1,18 +1,17 @@
>  # -*- coding: utf-8 -*-
>  # Generated by Django 1.9.8 on 2016-12-27 17:30
> +# Modified for Django 5.x on 2026-01-11
>  
>  
>  import datetime
>  
> -import pytz
> +from zoneinfo import ZoneInfo
>  from django.db import migrations, models
>  
> -PT = pytz.timezone("US/Pacific")
> -
>  
>  class Migration(migrations.Migration):
>  
> -    DEFAULT_DATETIME = PT.localize(datetime.datetime.now())
> +    DEFAULT_DATETIME = datetime.datetime.now(ZoneInfo("US/Pacific"))
>  
>      dependencies = [
>          ("registration", "0033_discount_used"),
> ```
> 
> Also on a sidenote, `influxdb` depends on `pytz` so despite removing it from the project's dependencies, it persists.

I recall an issue with how Django tracked times, and that I had some trouble with it, but nothing past that.  It was over a year ago?

p0lar-bear committed:
#24: Adding tzdata, converting pytz calls in migrations.

p0lar-bear committed:
24: Moving Trigram extension addition to new migration.

p0lar-bear committed:
#24: Tweaking envrc secrets file and settings templates, updating README.

.envrc.secrets is intended for direnv users, so I've made a settings template
for this use-case. Put secrets in that, configure other settings in the file.
Non-direnv users will be putting secrets into the settings.py file which is a
little risky, but what you gonna do?

p0lar-bear committed:
#24: Make direnv run pre-commit install, update docs.

rechner commented:
> CI is passing once again. There's just local tests breaking when running against a SQLite database. Kinda wanna take a closer look at the migrations and see about putting them back and making them work with Django 5.

We've relied on the Django json field type for a while which were only available in Postgres for a long time.  Perhaps that's changed in the current Django version, but it might just be easier to make Postgres a hard dependency.  Otherwise, it's expected for tests related to Square payments to fail when running in sqlite.

> I recall an issue with how Django tracked times, and that I had some trouble with it, but nothing past that. It was over a year ago?

Django migrates from `pytz` to `zoneinfo` in version 4.0, [per the release notes](https://docs.djangoproject.com/en/5.2/releases/4.0/#what-s-new-in-django-4-0).  `zoneinfo` is now part of the Python standard library since Python 3.9.  The pytz compatibility shim is removed completely from Django 5.0, so usages should be migrated like with your patch.

Notably this is only a change to the API for calculating timezones, and not really a database schema change.  Generating a migration is the right move.

LiruMouse commented on .github/workflows/django.yml:
```diff
@@ -49,26 +49,30 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Install uv
+        run: wget -qO- https://astral.sh/uv/install.sh | sh
```
https://docs.astral.sh/uv/guides/integration/github/
```suggestion
      - name: Install uv
        uses: astral-sh/setup-uv@v7
```

p0lar-bear committed:
#24: Use GitHub integration with uv in workflow

Co-authored-by: Liru Færs <lirusaito@gmail.com>

p0lar-bear commented:
> We've relied on the Django json field type for a while which were only available in Postgres for a long time. Perhaps that's changed in the current Django version, but it might just be easier to make Postgres a hard dependency. Otherwise, it's expected for tests related to Square payments to fail when running in sqlite.

@anadon I'm going to say nevermind to fixing those broken tests, then. I need to stand up Postgres in my dev environments instead of relying on Django's sqlite driver, especially when we start mucking about with payment. I just wanna update the development config template to reflect this and have it pull actual values from the .env file like the Docker/Travis config files.

anadon commented:
Got it.  I just went through a crash course at work fighting PostgreSQL.

On Fri, Jan 16, 2026, 8:32 PM Chris ***@***.***> wrote:

> *p0lar-bear* left a comment (furpocalypse/APIS#24)
> <https://github.com/furpocalypse/APIS/pull/24#issuecomment-3762465657>
>
> We've relied on the Django json field type for a while which were only
> available in Postgres for a long time. Perhaps that's changed in the
> current Django version, but it might just be easier to make Postgres a hard
> dependency. Otherwise, it's expected for tests related to Square payments
> to fail when running in sqlite.
>
> @anadon <https://github.com/anadon> I'm going to say nevermind to fixing
> those broken tests, then. I need to stand up Postgres in my dev
> environments instead of relying on Django's sqlite driver, especially when
> we start mucking about with payment. I just wanna update the development
> config template to reflect this and have it pull actual values from the
> .env file like the Docker/Travis config files.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/furpocalypse/APIS/pull/24#issuecomment-3762465657>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXYEBN2HG6U6Z5DB2WAAPL4HGGK5AVCNFSM6AAAAAB4MHJNFKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTONRSGQ3DKNRVG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

p0lar-bear committed:
Trying to find some middle ground between old and new. Realized that direnv
can read and process .env files, so I've ditched the .envrc.secrets file in
lieu of a .env.development template file. Decided to put the steps for the
database creation back in with a disclaimer that it's not going to work with
Debian systems.
